ethcore/engines/authority-round/src/lib.rs

```diff
@@ -122,6 +123,9 @@ pub struct AuthorityRoundParams {
 
 const U16_MAX: usize = ::std::u16::MAX as usize;
 
+/// The number of recent block hashes for which the gas limit override is memoized.
+const GAS_LIMIT_OVERRIDE_CACHE_CAPACITY: usize = 10;
```

Comment:
Yes, in POSDAO we'll have overrides for longer sequences of blocks.
But the main reason I added the cache is that with the current implementation in `verification.rs`, the gas limit is accessed three times for the current block and once for each uncle. So ideally, all of those values would still be cached.
If we expect blocks to rarely have more than 9 uncles, 10 should be fine. (And it takes very little memory anyway.)
Once a block has been verified and all of its nephews have been imported, it can drop out of the cache.

Reply:
Ok, that's a great explanation, thank you. Maybe extend the comment with that info? But I guess it depends on what happens with the rest of the code: if you manage to move it out of `verification` and into the engine, maybe this need to change.